src/java.base/share/classes/jdk/internal/loader/ClassLoaders.java

```diff
@@ -64,19 +64,20 @@ private static void setArchivedServicesCatalog(ClassLoader loader) {
     // Creates the built-in class loaders.
     static {
         ArchivedClassLoaders archivedClassLoaders = ArchivedClassLoaders.get();
+        // -Xbootclasspath/a or -javaagent with Boot-Class-Path attribute
+        String append = VM.getSavedProperty("jdk.boot.class.path.append");
+        URLClassPath bootAppendUcp = (append != null && !append.isEmpty())
+                ? new URLClassPath(append, true)
+                : null;
```

Comment:
Would you mind renaming this to bootUcp or bcp? The reason is that this is the boot loader's class path, not the append path. The system property has "append" in the name as it's communicating the value of -Xbootclasspath/a to the library code, this came about when -Xbootclasspath and -Xbootclasspath/p were removed.

Reply:
A full bcp would contain the path to the module image as the first path. This "append" property only contains the path specified in -Xbootclasspath/a. That's why I had bootAppendUcp as the variable name.
I've changed it to `bootUcp` per your suggestion.

Reply:
> A full bcp would contain the path to the module image as the first path. This "append" property only contains the path specified in -Xbootclasspath/a. That's why I had bootAppendUcp as the variable name. I've changed it to `bootUcp` per your suggestion.

There is no boot class path by default since JDK 9. If the boot loader is asked to load a class in a package that doesn't map to one of the modules defined to the boot loader then there is no class path to search. You've renamed it to bootUcp so I think we are good now.